mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td

```diff
@@ -531,7 +523,7 @@ def SYCLConstructorOp : SYCL_Op<"constructor", [CallOpInterface]> {
 
   let arguments = (ins
     Arg<SYCLMemref, "The SYCL object being created", [MemWrite]>:$Dst,
-    Variadic<ConstructorArgs>:$Args,  
+    Arg<Variadic<AnyType>, "The constructor arguments", [MemRead]>:$Args,
```

Comment:
You're right. We cannot ensure this, as the `accessor` constructor receives a reference to a `buffer`. Dropping that :+1:

Reply:
to be conservative, should we mark them memory write and read?
if we don't mark the memory effects, does it assume the worse? I worry that as the first argument is marked with memwrite, it would assume other args as no memory effects.

Reply:
I think it assumes the worst, but I can add it anyway